src/spider/storage/mysql/MySqlStorage.cpp

```diff
@@ -430,7 +467,16 @@ auto MySqlMetadataStorage::add_job(
                 };
             }
             Task const* task = task_option.value();
-            add_task(static_cast<MySqlConnection&>(conn), job_id_bytes, *task, TaskState::Ready);
+            auto const res = add_task(
+                    static_cast<MySqlConnection&>(conn),
+                    job_id_bytes,
+                    *task,
+                    TaskState::Ready
+            );
+            if (res.has_error()) {
+                static_cast<MySqlConnection&>(conn)->rollback();
+                return StorageErr{res.error().get_error(), "Cannot add task"};
+            }
```

Comment:
```suggestion
            if (auto const result = add_task(
                        static_cast<MySqlConnection&>(conn),
                        job_id_bytes,
                        *task,
                        TaskState::Ready
                );
                result.has_error())
            {
                static_cast<MySqlConnection&>(conn)->rollback();
                return StorageErr{result.error().get_error(), "Cannot add task"};
            }
```
* We'd prefer to use `xxx_result` or `result` in its full name.
* We'd prefer to put the `result` inside the if.